Api/Builder/QueueBuilderInterface.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Api\Builder;
+
+interface QueueBuilderInterface
```

Comment:
Like the idea of an interface for this - but I'm not sure of the name as it suggests that the class is responsible solely for constructing a queue. However, this service can handle batching, queuing, and potentially processing the items (depending on queue enablement configuration).

Perhaps something like `BatchQueueProcessor` might make more sense?

Reply:
I agree that this service is supposed to do "more" than just building the queue indeed. This issue is that historically, this is the queue model which handles the queue enablement configuration so we are in a weird spot where we use `addToQueue` in any case,  even if the queue is disabled. I agree to rename it but I think this is more a architecture issue that may need to be tackled one day 👍

Reply:
Yes we often find ourselves in "weird spots"! 😂